packages/extras/src/lib/components/controls/TransformControls/TransformControls.svelte

```diff
@@ -22,14 +22,17 @@
   type $$Slots = TransformControlsSlots
 
   export let autoPauseOrbitControls: $$Props['autoPauseOrbitControls'] = true
+  export let autoPauseTrackballControls: $$Props['autoPauseTrackballControls'] = true
   export let object: $$Props['object'] = undefined
 
   const { camera, renderer, invalidate, scene } = useThrelte()
 
-  const { orbitControls } = useControlsContext()
+  const { orbitControls, trackballControls } = useControlsContext()
```

Comment:
I'm not too familiar with this area of the codebase, but I wonder if we can avoid the repetition of code that occurs in the component in the current implementation.

Would there ever be a situation where `<TransformControls>` is in the context of _both_ `<OrbitControls>` _and_ `<TrackballControls>`? It seems unlikely to me. In this case, perhaps we can unify the logic?

This is not a binding comment since I am not sure I am right. I'd love @michealparks's feedback here also.

Reply:
yeah, the repetition feels icky but I don't yet know of a better way. My own rule of thumb is that repeating yourself up to 3 times is acceptable, but at the 4th time I try to find a better way 🤷 

Michael, if you have advice I'd be happy to follow it!

Reply:
Its not so much repetition itself which bothered me, but a concern that this may cause other issues, like overhead or unexpected behavior. Let's wait for @michealparks's input.

Reply:
Agreed that it feels awkward, but also agreed that we shouldn't overthink it until we're repeating ourselves many times. For now I think the type narrowing afforded by having two separate variables is more valuable. Having one `controls` store would make types more complicated.

Reply:
cool, gonna resolve this comment for now!